File: T9_Unlock.py
#!/usr/bin/env python3

import hashlib # imports the hashlib module for computing hashes
import sys # imports the sys module for accessing standard input and output

print("Enter your IMEI: ", end="") # prints a prompt without a newline
IMEI = input().strip() # reads a line from standard input and removes the trailing newline
hash = hashlib.sha1((IMEI + "simlock").encode()).hexdigest()[:8] # computes SHA1 hash of the IMEI concatenated with "simlock" and takes the first 8 characters
print(hash, end="") # prints the hash without a newline

File: T9_Unlock.ts
#!/usr/bin/env ts-node

// Import the required modules
import readline from 'readline';
import crypto from 'crypto';
import { Buffer } from 'buffer';

// Create a readline interface to read user input
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// Prompt the user to enter the IMEI number
rl.question('Enter your IMEI: ', (imei) => {
  // Append the word "simlock" to the IMEI number
  const input = imei + 'simlock';
  // Compute the SHA1 hash of the input
  const hash = crypto.createHash('sha1').update(input).digest('hex');
  // Cut the first eight characters of the hash
  const code = hash.slice(0, 8);
  // Print the code to the standard output
  console.log(code);
  // Close the readline interface
  rl.close();
});
